test/check-loopback

```diff
@@ -42,7 +54,8 @@ class TestLoopback(MachineCase):
         b.wait_visible('#login-fatal')
         self.assertIn("The server refused to authenticate 'admin' using password authentication, and no other supported authentication methods are available.", b.text('#login-fatal'));
 
-        self.allow_journal_messages("cockpit-polkit: couldn't respond to polkit daemon: GDBus\\.Error:org\\.freedesktop\\.PolicyKit1\\.Error\\.Failed: No session for cookie")
+        self.allow_journal_messages("cockpit-polkit: couldn't respond to polkit daemon: GDBus\\.Error:org\\.freedesktop\\.PolicyKit1\\.Error\\.Failed: No session for cookie",
+                                    ".*server offered unsupported authentication methods: public-key.*")
```

Comment:
On 12/02/2015 05:29 AM, Marius Vollmer wrote:
> In test/check-loopback

> The message really means "server offered /only/ unsupported methods",
> which isn't true for Debian: it has offered the password method, which
> we support, but there we didn't try because cockpit-ws didn't have a
> password to use.

I don't think that's true, the test disables password authentication, so 
the server doesn't offer it. It only offered public-key which we don't 
support yet on debian so the message seems right to me.

>
> This code is inconsistent re password and public-key there: public-key
> is always considered supported (when compiled in), even if there is no
> key in cockpit-ws (or the agent) that could be used. On Fedora, where we
> do support public-key, we thus never see this message.

It is true that there is an inconsistency between public-key and 
password. not-supported should work correctly with public-key but we 
don't check that we actually have a valid agent with keys so we never 
respond with not-provided when there are none in the same way as we do 
when no password is supplied. I'm not sure that's worth fixing though as 
querying the agent here is not trivial but if everyone thinks that is 
worth doing I can take shot at it.


>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockpit-project/cockpit/pull/3202/files#r46413264>.
>